core/src/gas.rs

```diff
@@ -301,6 +301,44 @@ pub trait CountersOwner {
     fn set_gas_left(&mut self, gas_left: GasLeft);
 }
 
+/// Provide functionality to safely make gas refund after charge,
+/// it checks, that it's not refunded more gas than has been charged.
+pub struct GasRefunder {
+    amount: u64,
+}
+
+/// [GasRefunder] error, in case somebody tries to refund more than was charged.
+#[derive(Debug, derive_more::Display)]
+#[display(fmt = "Trying to refund {charged}, but {refunded} was charged")]
+pub struct RefundError {
+    charged: u64,
+    refunded: u64,
+}
+
+impl GasRefunder {
+    /// Charge gas.
+    pub fn charge<C: CountersOwner>(counters: &mut C, amount: u64) -> Result<Self, ChargeError> {
```

Comment:
```suggestion
    pub fn charge_gas_if_enough<C: CountersOwner>(counters: &mut C, amount: u64) -> Result<Self, ChargeError> {
```
To make it more clear

Reply:
done